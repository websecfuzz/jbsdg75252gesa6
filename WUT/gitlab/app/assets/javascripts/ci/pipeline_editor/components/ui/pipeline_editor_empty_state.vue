<script>
import { GlButton, GlSprintf, GlEmptyState } from '@gitlab/ui';
import { __ } from '~/locale';
import PipelineEditorFileNav from '~/ci/pipeline_editor/components/file_nav/pipeline_editor_file_nav.vue';

export default {
  components: {
    GlButton,
    GlSprintf,
    GlEmptyState,
    PipelineEditorFileNav,
  },
  i18n: {
    title: __('Optimize your workflow with CI/CD Pipelines'),
    body: __(
      'Create a new %{codeStart}.gitlab-ci.yml%{codeEnd} file at the root of the repository to get started.',
    ),
    btnText: __('Configure pipeline'),
    externalCiNote: __("This project's pipeline configuration is located outside this repository"),
    externalCiInstructions: __(
      'To edit the pipeline configuration, you must go to the project or external site that hosts the file.',
    ),
  },
  inject: {
    emptyStateIllustrationPath: {
      default: '',
    },
    usesExternalConfig: {
      default: false,
      type: Boolean,
      required: false,
    },
  },
  methods: {
    createEmptyConfigFile() {
      this.$emit('createEmptyConfigFile');
    },
  },
};
</script>
<template>
  <div>
    <pipeline-editor-file-nav v-on="$listeners" />
    <gl-empty-state
      v-if="usesExternalConfig"
      :title="$options.i18n.externalCiNote"
      :description="$options.i18n.externalCiInstructions"
      :svg-path="emptyStateIllustrationPath"
    />

    <gl-empty-state v-else :title="$options.i18n.title" :svg-path="emptyStateIllustrationPath">
      <template #description>
        <gl-sprintf :message="$options.i18n.body">
          <template #code="{ content }">
            <code>{{ content }}</code>
          </template>
        </gl-sprintf>
      </template>
      <template #actions>
        <gl-button
          variant="confirm"
          class="gl-mt-3"
          data-testid="create-new-ci-button"
          @click="createEmptyConfigFile"
        >
          {{ $options.i18n.btnText }}
        </gl-button>
      </template>
    </gl-empty-state>
  </div>
</template>
