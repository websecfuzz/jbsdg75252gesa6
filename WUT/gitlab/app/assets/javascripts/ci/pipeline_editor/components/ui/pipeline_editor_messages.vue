<script>
import { GlAlert } from '@gitlab/ui';
import { getParameterValues, removeParams } from '~/lib/utils/url_utility';
import { __, s__ } from '~/locale';
import {
  COMMIT_FAILURE,
  DEFAULT_FAILURE,
  LOAD_FAILURE_UNKNOWN,
  PIPELINE_FAILURE,
} from '../../constants';
import CodeSnippetAlert from '../code_snippet_alert/code_snippet_alert.vue';
import {
  CODE_SNIPPET_SOURCE_URL_PARAM,
  CODE_SNIPPET_SOURCES,
} from '../code_snippet_alert/constants';

export default {
  components: {
    GlAlert,
    CodeSnippetAlert,
  },

  errors: {
    [COMMIT_FAILURE]: s__('Pipelines|The GitLab CI configuration could not be updated.'),
    [DEFAULT_FAILURE]: __('Something went wrong on our end.'),
    [LOAD_FAILURE_UNKNOWN]: s__('Pipelines|The CI configuration was not loaded, please try again.'),
    [PIPELINE_FAILURE]: s__('Pipelines|There was a problem with loading the pipeline data.'),
  },
  props: {
    failureType: {
      type: String,
      required: false,
      default: null,
    },
    failureReasons: {
      type: Array,
      required: false,
      default: () => [],
    },
    showFailure: {
      type: Boolean,
      required: false,
      default: false,
    },
  },
  data() {
    return {
      codeSnippetCopiedFrom: '',
    };
  },
  computed: {
    failure() {
      const { errors } = this.$options;

      return {
        text: errors[this.failureType] ?? errors[DEFAULT_FAILURE],
        variant: 'danger',
      };
    },
  },
  created() {
    this.parseCodeSnippetSourceParam();
  },
  methods: {
    dismissCodeSnippetAlert() {
      this.codeSnippetCopiedFrom = '';
    },
    dismissFailure() {
      this.$emit('hide-failure');
    },
    parseCodeSnippetSourceParam() {
      const [codeSnippetCopiedFrom] = getParameterValues(CODE_SNIPPET_SOURCE_URL_PARAM);
      if (codeSnippetCopiedFrom && CODE_SNIPPET_SOURCES.includes(codeSnippetCopiedFrom)) {
        this.codeSnippetCopiedFrom = codeSnippetCopiedFrom;
        window.history.replaceState(
          {},
          document.title,
          removeParams([CODE_SNIPPET_SOURCE_URL_PARAM]),
        );
      }
    },
  },
};
</script>

<template>
  <div>
    <code-snippet-alert
      v-if="codeSnippetCopiedFrom"
      :source="codeSnippetCopiedFrom"
      class="gl-mb-5"
      @dismiss="dismissCodeSnippetAlert"
    />
    <gl-alert
      v-if="showFailure"
      :variant="failure.variant"
      class="gl-mb-5"
      @dismiss="dismissFailure"
    >
      {{ failure.text }}
      <ul v-if="failureReasons.length" class="gl-mb-0">
        <li v-for="reason in failureReasons" :key="reason">{{ reason }}</li>
      </ul>
    </gl-alert>
  </div>
</template>
