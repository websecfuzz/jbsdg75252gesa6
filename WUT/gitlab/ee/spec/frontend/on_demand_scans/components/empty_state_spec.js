import { mount } from '@vue/test-utils';
import { GlEmptyState } from '@gitlab/ui';
import EmptyState from 'ee/on_demand_scans/components/empty_state.vue';

describe('EmptyState', () => {
  let wrapper;

  // Finders
  const findGlEmptyState = () => wrapper.findComponent(GlEmptyState);

  // Helpers
  const defaultGlEmptyStateProp = (prop) => GlEmptyState.props[prop].default;

  const createComponent = (propsData = {}) => {
    wrapper = mount(EmptyState, {
      provide: {
        newDastScanPath: '/on_demand_scans/new',
        emptyStateSvgPath: '/empty/state/svg/path',
      },
      propsData,
    });
  };

  it('renders properly', () => {
    createComponent();

    expect(wrapper.element).toMatchSnapshot();
  });

  it('passes custom title down to GlEmptyState', () => {
    const title = 'Custom title';
    createComponent({
      title,
    });

    expect(findGlEmptyState().props('title')).toBe(title);
  });

  it('passes custom text down to GlEmptyState', () => {
    const text = 'Custom text';
    createComponent({
      text,
    });

    expect(findGlEmptyState().text()).toMatch(text);
  });

  it('does not pass primary props when no-primary-button is true', () => {
    createComponent({
      noPrimaryButton: true,
    });

    expect(findGlEmptyState().props('primaryButtonLink')).toBe(
      defaultGlEmptyStateProp('primaryButtonLink'),
    );
    expect(findGlEmptyState().props('primaryButtonLink')).toBe(
      defaultGlEmptyStateProp('primaryButtonLink'),
    );
  });
});
