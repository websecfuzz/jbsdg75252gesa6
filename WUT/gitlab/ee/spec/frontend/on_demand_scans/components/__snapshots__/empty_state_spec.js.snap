// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState renders properly 1`] = `
<section
  class="gl-empty-state gl-flex gl-flex-col gl-text-center"
>
  <div
    class="gl-max-w-full"
  >
    <img
      alt=""
      class="gl-dark-invert-keep-hue gl-max-w-full"
      height="144"
      src="/empty/state/svg/path"
    />
  </div>
  <div
    class="gl-empty-state-content gl-m-auto gl-mx-auto gl-my-0 gl-p-5"
    data-testid="gl-empty-state-content"
  >
    <h1
      class="gl-leading-36 gl-mb-0 gl-mt-0 gl-text-size-h-display h4"
    >
      On-demand scans
    </h1>
    <p
      class="gl-mb-0 gl-mt-4 gl-text-subtle"
    >
      On-demand scans run outside of DevOps cycle and find vulnerabilities in your projects.
      <a
        class="gl-link"
        href="/help/user/application_security/dast/on-demand_scan"
      >
        Learn more
      </a>
      .
    </p>
    <div
      class="gl-flex gl-flex-wrap gl-justify-center gl-mt-5"
    >
      <a
        class="btn btn-confirm btn-md gl-button gl-mb-3 gl-mx-2"
        href="/on_demand_scans/new"
      >
        <span
          class="gl-button-text"
        >
          New scan
        </span>
      </a>
    </div>
  </div>
</section>
`;
